gen3-integration-tests/tests/test_graph_submit_and_query.py

```diff
@@ -60,42 +95,436 @@ def test_submit_query_and_delete_records(self):
-            sd_tools.delete_all_test_records()
-
-    # TODO add tests:
-    # submit node unauthenticated
-    # submit node without parent
-    # test with_path_to - first to last node
-    # test with_path_to - last to first node
-    # submit data node with consent codes
+                self.sd_tools.delete_records(
+                    [record.unique_id for record in new_records]
+                )
+            self.sd_tools.delete_all_test_records()
+
+    def test_submit_node_unauthenticated(self):
+        """
+        Scenario: Submit node unauthenticated
+        Steps:
+            1. Generate an expired token
+            2. Create a node with expired token
+            3. Creating node should fail
+        """
+        # Generate an expired token
+        res = create_access_token(
+            pytest.namespace, "fence", "1", "cdis.autotest@gmail.com"
+        )
+
+        res = res.splitlines()[-1].strip()
+        auth_header = {
+            "Accept": "application/json",
+            "Authorization": f"bearer {res}",
+            "Content-Type": "application/json",
+        }
+
+        # Create a node with expired token
+        first_node = self.sd_tools.test_records[self.sd_tools.submission_order[0]]
+        response = requests.put(
+            url=pytest.root_url + "/api/v0/submission/jnkns/jenkins",
+            data=json.dumps(first_node.props),
+            headers=auth_header,
+        )
+        assert (
+            response.status_code == 401
+        ), f"Failed to delete record. Response: {response}"
+
+    @pytest.mark.graph_query
+    def test_submit_record_without_parent(self):
+        """
+        Scenario: Submit record without parent
+        Steps:
+            1. Create a record using sheepdog. Verify record is present.
+            2. Perform a query using an invalid project_id.
+            3. Validate length of fields returned for node name is 0
+            4. Delete the record created. Verify record is deleted.
+        """
+        # Verify parent node does not exist
+        parent_node = self.sd_tools.test_records[self.sd_tools.submission_order[0]]
+        parent_response = self.sd_tools.query_record_fields(parent_node)
+        # Validate length of fields returned for node name is 0
+        if len(parent_response["data"][parent_node.node_name]) != 0:
+            logger.error(
+                "Found fields for {}. Response : {}".format(
+                    parent_node.node_name, parent_response
+                )
+            )
+            raise
+
+        # Attempt to add second node which has a dependency on another node
+        second_node = self.sd_tools.get_record_with_parent()
+        try:
+            logger.info(self.sd_tools.submit_record(record=second_node))
+        except requests.exceptions.HTTPError as e:
+            if e.response.status_code != 400:
+                logger.error(f"Expected 400 status code not found. Response: {e}")
+                raise
```

Comment:
This won't work, we need to make sure there was an error, not just check the nature of the error if there was one. This wouldn't catch an unexpected 200 successful response.
Look into `with pytest.raises` to fix this block

Reply:
have moved the try catch block to pytest.raises